app/f0_serialcom/main.c

```diff
@@ -0,0 +1,97 @@
+/* STM32F0xx Serial terminal example
+ *
+ * Jeff Ciesielski <jeff.ciesielski@gmail.com>
+ */
+
+/* NOTE: This example assumes that you are attaching a serial
+ * terminal to USART1 (PA9/PA10) via something like an FTDI cable */
+
+#include <FreeRTOS.h>
+#include <task.h>
+
+#include <stm32f0xx_misc.h>
+#include <blinky.h>
+
+#include <stdio.h>
+#include <usart.h>
+#include <shell.h>
+#include <alloca.h>
+
+void vApplicationTickHook(void)
+{
+}
+
+void vApplicationStackOverflowHook(xTaskHandle pxTask, signed char *pcTaskName)
+{
+	(void)pxTask;
+	(void)pcTaskName;
+}
+
+void HardFault_Handler(void)
+{
+	while(1)
+		;
+}
+
+static void tasks_help(void)
+{
+	printf("tasks - show runtime task information\n");
+}
+
+static void tasks_handler(int argc, char **argv)
+{
+	int task_count = uxTaskGetNumberOfTasks();
+	char *tbuf = alloca(task_count * 40);
+	printf("Name         State     Prio    Stack  Num\n");
+	printf("-----------------------------------------\n");
+	vTaskList((signed char*)tbuf);
```

Comment:
Same signed-char curiosity here. Reason why I mention it is that 'char' != 'signed char', despite the fact that the equality does hold for all non-char integral types. (C is weird that way...)